chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -150,7 +150,10 @@ def self.unclaimed(node, include_mounted=false)
               %x{lsblk #{d.name.gsub(/!/, "/")} --noheadings --output MOUNTPOINT | grep -q -v ^$}
               next if $?.exitstatus == 0
             end
-            d.fixed and not d.claimed?
+            # skip claimed disks and multipath devices held by holders
+            # but include both fixed disks and multipath devices
```

Comment:
This is certainly getting easier to understand, but the comment doesn't really add much other than just repeat what the code says.  Ideally the comment should explain *why* the code is written that way, not simply what it does.  I would suggest introducing temporary variables to highlight that whilst some devices will *never* be claimable by their inherent nature, whereas others will be claimable unless they're already claimed.  For example:

    device_type_claimable = d.fixed || d.multipath?
    in_use = d.claimed? || d.held_by_multipath?
    device_type_claimable && !in_use

For me, splitting the logic into these two categories makes it clear enough that it essentially becomes self-documenting and doesn't need a comment any more.  Does that make sense?